content/en/docs/guides/flux-e2e.md

```diff
@@ -0,0 +1,330 @@
+
+A [HelmRelease][HelmRelease API] is a composition of a chart, the chart values (parameters to the chart), and any inputs like secrets or config maps that are used to compose the values. Declaring a HelmRelease will cause the Helm Controller to perform an install using the Helm client libraries. Any changes made to the HelmRelease will trigger the Helm Controller to perform an upgrade to actuate those changes. You can find more information about HelmReleases [here][HelmRelease Guide] and more general info about Helm for Flux users [here][Helm Use Case].
+
+
+### 12. HelmRepository and HelmChart (Sources for Helm)
+
+A Helm Repository is the native and preferred source for Helm. The Helm Controller works in tandem with the Source Controller to provide a [HelmRepository API][] that collects the correct release version from the helm repo and republishes its data as a [HelmChart][HelmChart API] artifact (another .tar.gz).
+
+The helm repo itself is represented internally in the Source Controller as a YAML index of all releases, including any charts in the repository.
+
+### 13. GitRepository and HelmChart (Alternative Sources for Helm)
+
+GR can be used as a source for Helm Release. The Git repo is not a native storage format for helm and there are some idiosyncrasies when you’re using Helm Controller with a Git repository source. You can use a GitRepository as a source, but best practice is to limit it to 1:1 (don’t do mono repo) - bad idea to create a repo with 400 helm charts. The problem is that git repo sources are tgz files end up with lots of artifacts pulled each time (overloading). Orange juice analogy here?
+
+So you have lots of tools at your disposal for making sources narrowly scoped, and if you will use them all, you can avoid any potential issues stemming from Helm Controller accidentally pulling in resources and causing source controller to repackage them again, when you did not need to include them in the chart.
+
+Whatever it says it just needs to get the message across that the Source is really a .tgz at any point in time, which can't be partially downloaded, and the Helm Chart is also a .tgz, so if you want this to perform well, you need to keep irrelevant things out of both TGZ files (and this unfortunately means you will have a lot more GitRepository resources than you wanted, at least for now. Not sure how this can be improved without drastic architectural changes that seem unlikely.)
+
+It's probably important to get across also that whatever winds up in the Helm Chart .tgz will actually be loaded into memory by Helm, and it has no way to know which files are not important. So while you can get away with reusing one GitRepository up to a point, you really need to know for sure that what goes into the HelmChart is not littered with unnecessary files more than what's needed.
+
+Links/resources:
+
+TODO: this all belongs in one of the Helm guides
+
+### 14. Notifications Part 1 - Notification Providers
```

Comment:
I would avoid using "part <num>" in the titels, as they add little (contextual) value.